frontend/src/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,236 @@
+      }
+    }
+  }
+
+  private setWidgetValue = (source: Source): void => {
+    const { value } = this.state
+    const { element, widgetMgr } = this.props
+    const defaultValue: number = element.get("default")
+    const widgetId: string = element.get("id")
+    const format: string = element.get("format")
+
+    const valueToBeSaved = value === "" ? defaultValue.toString() : value
+    const formattedValue = format
+      ? sprintf(format, valueToBeSaved)
+      : valueToBeSaved
+
+    if (this.strIsInt(valueToBeSaved)) {
+      widgetMgr.setIntValue(widgetId, parseInt(formattedValue), source)
+    } else {
+      widgetMgr.setFloatValue(widgetId, parseFloat(formattedValue), source)
+    }
+
+    this.setState({
+      dirty: false,
+      value: formattedValue,
+    })
+  }
+
+  private onBlur = (): void => {
+    if (this.state.dirty) {
+      this.setWidgetValue({ fromUi: true })
+    }
+  }
+
+  private onChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const { value } = e.target
+
+    this.setState({
+      dirty: true,
+      value,
+    })
+  }
+
+  private onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+    const { key } = e
+
+    switch (key) {
+      case "ArrowUp":
+        e.preventDefault()
+
+        this.modifyValueUsingStep("increment")()
+        break
+      case "ArrowDown":
+        e.preventDefault()
+
+        this.modifyValueUsingStep("decrement")()
+        break
+    }
+  }
+
+  private onKeyPress = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+    if (e.key === "Enter" && this.state.dirty) {
+      this.setWidgetValue({ fromUi: true })
+    }
+  }
+
+  private modifyValueUsingStep = (
+    modifier: "increment" | "decrement"
+  ): any => (): void => {
+    const { element } = this.props
+    const step = this.getStep()
+    const value = this.getValue()
+    const format = element.get("format")
+
+    switch (modifier) {
+      case "increment":
+        this.setState({
+          dirty: true,
+          value: sprintf(format, value + step),
+        })
+        break
+      case "decrement":
+        this.setState({
+          dirty: true,
+          value: sprintf(format, value - step),
+        })
+        break
+    }
+  }
+
+  private startEnhancer = () => {
+    return (
+      <span
+        className="startEnhancer"
+        onClick={this.modifyValueUsingStep("decrement")}
+      >
+        -
+      </span>
+    )
+  }
+
+  private endEnhancer = () => {
+    return (
+      <span
+        className="endEnhancer"
+        onClick={this.modifyValueUsingStep("increment")}
+      >
+        +
+      </span>
+    )
+  }
+
+  private getEnhancersStyle = () => ({
+    cursor: "pointer",
+    padding: 0,
+  })
+
+  public render = (): React.ReactNode => {
+    const { element, width, disabled } = this.props
+    const { value, dirty } = this.state
+
+    const label: string = element.get("label")
+    const style = { width }
+    const inputValue = value || ""
```

Comment:
if `value` is 0 then `inputValue` will be ""

Reply:
The attribute value it's a string, but i will remove this line as it is no longer needed

Reply:
removed